### PR TITLE
[Minor Breaking API] Make deallocation queues more reliable

### PR DESCRIPTION
**Pull request by Adlai-Holler**
In order to be sure we've released our local reference before the queue releases its reference, we need to pass a double-pointer.

Minor breaking API changes in this diff:
- `ASPerformBackgroundDeallocation(x) -> ASPerformBackgroundDeallocation(&x)`
- Swift: `ASDeallocQueue.sharedDeallocationQueue() -> ASDeallocQueue.sharedDeallocationQueue`
- `ASPerformMainThreadDeallocation(x) -> ASPerformMainThreadDeallocation(&x)`

### CODE REVIEW

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/08643165310c3e6b16915e70af150e84/raw/52d247d2fee9f0f4702564365324f4316ed56ce0/log.txt)

**Comment by appleguy** [comment]
Heh, good finds. I’m on my phone at the moment and not totally confident in my review, but I will look into it with more detail when I can (land if you get another though!)

File: Source/ASDisplayNodeExtras.h [hunk 1]
```diff
@@ -35,7 +35,7 @@
 #endif
 
 /// For deallocation of objects on the main thread across multiple run loops.
-extern void ASPerformMainThreadDeallocation(_Nullable id object);
+extern void ASPerformMainThreadDeallocation(id _Nullable __strong * _Nonnull objectPtr);
```
Comment (nguyenhuy) [inline Source/ASDisplayNodeExtras.h]: Didn't know this is legit. Learning every day!

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
 - [ASNetworkImageNode] New delegate callback to tell the consumer whether the image was loaded from cache or download. [Adlai Holler](https://github.com/Adlai-Holler)
 - [Layout] Fixes a deadlock in layout. [#638](https://github.com/TextureGroup/Texture/pull/638) [Garrett Moon](https://github.com/garrettmoon)
 - Updated to be backwards compatible with Xcode 8. [Adlai Holler](https://github.com/Adlai-Holler)
+- Fixed cases where main-thread-deallocation doesn't work as expected [Adlai Holler](https://github.com/Adlai-Holler)
```
Comment (nguyenhuy) [inline CHANGELOG.md]: Please add a note mentioning the API change.

**Comment by nguyenhuy** [comment]
Tested with Pinterest app and confirm that this PR fixes #607.

**Pull request merged**